File: sources/lua/04_array_sum/main.lua
-- task 04 array_sum — expected output: 499999500000
-- build: none (interpreted)    run: lua main.lua (PUC Lua 5.4)    also runs on LuaJIT: luajit main.lua

local n = 1000000
local array = {}

for i = 1, n do
    array[i] = i - 1
end

local total = 0
for i = 1, n do
    total = total + array[i]
end

print(total)

File: sources/ruby/04_array_sum/main.rb
# task 04 array_sum — expected output: 499999500000
# build: none (interpreted)    run: ruby main.rb (cruby) | ruby --yjit main.rb (cruby+yjit) | jruby main.rb (jruby, needs Java 25)
# The stock Windows CRuby build has no YJIT: `ruby --yjit` warns "Ruby was built without YJIT support".

n = 1_000_000
array = Array.new(n, 0)

i = 0
while i < n
  array[i] = i
  i += 1
end

total = 0
i = 0
while i < n
  total += array[i]
  i += 1
end

puts total
